### DISCUSSION TITLE
grblHAL support?

### DISCUSSION THREAD

**Discussion by stevempotter** [winder/Universal-G-Code-Sender#1859] [Help]
I am running grblHAL on the Teensy4.1 MCU. Will the UGCS be supporting that any time soon?

**Comment by Hanier** [comment]
Bump... Any thoughts on bringing UGS (myFAV) up to speed on grblHAL?

**Reply by breiler** [reply]
It should run fine in compatibility mode:
https://github.com/grblHAL/core/wiki/Compatibility-level

Is there any new grblHAL feature you are missing?

**Reply by Hanier** [reply]
Hi,

I really like the way & feel of UGS, and hoped to use it moving forward.  I am bringing up a new system; actually upgrading my ArduinoUNO w/grbl V1.1h to a new board using PhilB's Teensy4.1 break-out-board.

I see a few things that threw me.
1. In some code intended to peck drill, I noticed UGS showing concerns about G98 and G83.
2. Too, it reports: "Connected to GRBL 1.1f." <<<(I think that might come from the new grblHAL board though)
3. When making the connection, there is no selection for grblHAL.

All of this leaving me wondering if everything will be alright.  I did not want to 'tone down' using compatibility modes - that's why I am doing the upgrade to my system. ;-)  Am I going to run into issues?

Regards, David
P.S.  I attached a photo showing UGS and the warnings, but I am not completely sure it worked.

![UGS_shows_Concerns](https://github.com/winder/Universal-G-Code-Sender/assets/9752446/47c19983-9ed9-4baf-835c-98a44207aba2)
[DrillPlay_Rgrbl.txt](https://github.com/user-attachments/files/15988561/DrillPlay_Rgrbl.txt)

**Reply by Hanier** [reply]
Ahh.  It did work.  Now I see the photo.
